From a8d55e7f0e0e431a94d199a793935c27264453a0 Mon Sep 17 00:00:00 2001
From: Daniel Charles <daniel.charles@intel.com>
Date: Wed, 19 Jul 2017 13:16:15 -0700
Subject: [PATCH] va_enc_h264: fix union/struct typo to silence warning

This is a backport to fix API build warning.  The final solution
will land later upstream.

ChromeOS specific patch to avoid warnings when building browser
against system installed libva headers

Signed-off-by: Daniel Charles <daniel.charles@intel.com>
---
 va/va_enc_h264.h | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/va/va_enc_h264.h b/va/va_enc_h264.h
index 2e7eb8d..4d0c1df 100644
--- a/va/va_enc_h264.h
+++ b/va/va_enc_h264.h
@@ -545,8 +545,8 @@ typedef struct _VAEncMacroblockParameterBufferH264 {
     union {
         /** @name Data for intra macroblock */
         /**@{*/
-        struct {
-            union {
+        union {
+            struct {
                 /**
                  * \brief Flag specified to override MB neighbour
                  * availability bits from VME stage.
@@ -573,8 +573,9 @@ typedef struct _VAEncMacroblockParameterBufferH264 {
 
         /** @name Data for inter macroblock */
         /**@{*/
-        struct {
-            union {
+        union {
+            struct {
+                unsigned int reserved;
             } bits;
             unsigned int value;
         } inter_fields;
-- 
2.9.4

